src/model.h

```diff
@@ -31,6 +31,73 @@
 #include "scheduler.h"
 #include "status.h"
 
+namespace triton { namespace core {
+struct ModelIdentifier {
+  ModelIdentifier(const std::string& model_namespace, const std::string& name)
+   : namespace_(model_namespace), name_(name) {}
+
+  bool operator<(const ModelIdentifier& rhs) const
+  {
+    if (namespace_ == rhs.namespace_) {
+      return (name_ < rhs.name_);
+    } else {
+      return (namespace_ < rhs.namespace_);
+    }
+  }
+
+  bool operator==(const ModelIdentifier& rhs) const
+  {
+    return ((namespace_ == rhs.namespace_) && (name_ == rhs.name_));
+  }
+
+  bool operator!=(const ModelIdentifier& rhs) const
+  {
+    return (!(namespace_ != rhs.namespace_) || !(name_ != rhs.name_));
```

Comment:
I am wondering if you mean `(namespace_ != rhs.namespace_) || (name_ != rhs.name_)`? So, `(lhs == rhs) == !(lhs != rhs)`

Reply:
Yes, missed the cleanup after expanding "not" operation